js/noms/src/dataset.js

```diff
@@ -34,20 +35,38 @@ export default class Dataset {
     return this._database;
   }
 
+  /** ID returns the name of this Dataset. */
   get id(): string {
     return this._id;
   }
 
   // TODO: This should return Promise<Ref<Commit> | null>.
+  /**
+   * HeadRef returns the Ref of the current head Commit, which contains the
+   * current root of the Dataset's value tree.
+   * If the named Dataset doesn't exist in this._database, the returned Promise may resolve to null.
```

Comment:
I would not mention `this._database`. Maybe "doesn't exist in the underlying database"?